quinn-proto/src/endpoint.rs

```diff
@@ -395,7 +395,11 @@ impl Endpoint {
             return Err(ConnectError::UnsupportedVersion);
         }
 
-        let remote_id = RandomConnectionIdGenerator::new(MAX_CID_SIZE).generate_cid();
+        let remote_id = config
+            .initial_dst_cid_generator
+            .lock()
```

Comment:
It would probably be better to change the type of `initial_dst_cid_generator` to `Arc<dyn Fn() -> ConnectionId>`, since `generate_cid` is the only member of `ConnectionIdGenerator` that is used here.